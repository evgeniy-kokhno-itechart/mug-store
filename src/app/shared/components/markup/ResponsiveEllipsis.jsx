import React from 'react';
import LinesEllipsis from 'react-lines-ellipsis';
import responsiveHOC from 'react-lines-ellipsis/lib/responsiveHOC';
import { PropTypes } from 'prop-types';

const RespEllipsis = responsiveHOC()(LinesEllipsis);

const ResponsiveEllipsis = ({
  text, maxLine, ellipsis, basedOn, customClasses,
}) => (
  // inline style here to fix issue https://github.com/xiaody/react-lines-ellipsis/issues/59
  <RespEllipsis
    className={customClasses}
    text={text}
    maxLine={maxLine}
    ellipsis={ellipsis}
    trimRight
    basedOn={basedOn}
    style={{ whiteSpace: 'pre-wrap' }}
  />
);

ResponsiveEllipsis.propTypes = {
  text: PropTypes.string,
  maxLine: PropTypes.string,
  ellipsis: PropTypes.string,
  basedOn: PropTypes.oneOf(['letters', 'words']),
  customClasses: PropTypes.string,
};

ResponsiveEllipsis.defaultProps = {
  text: '',
  maxLine: '1',
  ellipsis: '...',
  basedOn: 'words',
  customClasses: '',
};

export default ResponsiveEllipsis;
