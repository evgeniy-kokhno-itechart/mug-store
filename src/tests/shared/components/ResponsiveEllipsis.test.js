/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ResponsiveEllipsis from '../../../app/shared/components/markup/ResponsiveEllipsis';

configure({ adapter: new Adapter() });

describe('<ResponsiveEllipsis />', () => {
  let ResponsiveEllipsisWrapper;

  beforeEach(() => {
    ResponsiveEllipsisWrapper = shallow(<ResponsiveEllipsis />);
  });

  it('renders properly with default props', () => {
    expect(ResponsiveEllipsisWrapper).toMatchSnapshot();
  });

  it('renders properly with text provided', () => {
    ResponsiveEllipsisWrapper.setProps({ text: 'Lorem ipsum dolor sit amet' });
    expect(ResponsiveEllipsisWrapper).toMatchSnapshot();
  });

  it('renders properly with long text and maxLine provided', () => {
    ResponsiveEllipsisWrapper.setProps({
      text:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc rutrum justo sit amet suscipit tempor. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Phasellus magna lacus, egestas vitae urna sed, pulvinar blandit sem. Donec ultrices dolor at neque convallis, ut fringilla quam tincidunt.',
      maxLine: '2',
    });
    expect(ResponsiveEllipsisWrapper).toMatchSnapshot();
  });

  it('renders properly with text and ellipsis provided', () => {
    ResponsiveEllipsisWrapper.setProps({ text: 'Lorem ipsum dolor sit amet', ellipsis: '***' });
    expect(ResponsiveEllipsisWrapper).toMatchSnapshot();
  });

  it('renders properly with text and basedOn words provided', () => {
    ResponsiveEllipsisWrapper.setProps({ text: 'Lorem ipsum dolor sit amet', basedOn: 'words' });
    expect(ResponsiveEllipsisWrapper).toMatchSnapshot();
  });

  it('renders properly with text and basedOn letters provided', () => {
    ResponsiveEllipsisWrapper.setProps({ text: 'Lorem ipsum dolor sit amet', basedOn: 'letters' });
    expect(ResponsiveEllipsisWrapper).toMatchSnapshot();
  });
});
