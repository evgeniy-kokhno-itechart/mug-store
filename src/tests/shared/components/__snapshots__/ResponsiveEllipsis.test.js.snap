// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponsiveEllipsis /> renders properly with default props 1`] = `
<Responsive(LinesEllipsis)
  basedOn="words"
  ellipsis="..."
  innerRef={[Function]}
  maxLine="1"
  style={
    Object {
      "whiteSpace": "pre-wrap",
    }
  }
  text=""
  trimRight={true}
/>
`;

exports[`<ResponsiveEllipsis /> renders properly with long text and maxLine provided 1`] = `
<Responsive(LinesEllipsis)
  basedOn="words"
  ellipsis="..."
  innerRef={[Function]}
  maxLine="2"
  style={
    Object {
      "whiteSpace": "pre-wrap",
    }
  }
  text="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc rutrum justo sit amet suscipit tempor. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Phasellus magna lacus, egestas vitae urna sed, pulvinar blandit sem. Donec ultrices dolor at neque convallis, ut fringilla quam tincidunt."
  trimRight={true}
/>
`;

exports[`<ResponsiveEllipsis /> renders properly with text and basedOn letters provided 1`] = `
<Responsive(LinesEllipsis)
  basedOn="letters"
  ellipsis="..."
  innerRef={[Function]}
  maxLine="1"
  style={
    Object {
      "whiteSpace": "pre-wrap",
    }
  }
  text="Lorem ipsum dolor sit amet"
  trimRight={true}
/>
`;

exports[`<ResponsiveEllipsis /> renders properly with text and basedOn words provided 1`] = `
<Responsive(LinesEllipsis)
  basedOn="words"
  ellipsis="..."
  innerRef={[Function]}
  maxLine="1"
  style={
    Object {
      "whiteSpace": "pre-wrap",
    }
  }
  text="Lorem ipsum dolor sit amet"
  trimRight={true}
/>
`;

exports[`<ResponsiveEllipsis /> renders properly with text and ellipsis provided 1`] = `
<Responsive(LinesEllipsis)
  basedOn="words"
  ellipsis="***"
  innerRef={[Function]}
  maxLine="1"
  style={
    Object {
      "whiteSpace": "pre-wrap",
    }
  }
  text="Lorem ipsum dolor sit amet"
  trimRight={true}
/>
`;

exports[`<ResponsiveEllipsis /> renders properly with text provided 1`] = `
<Responsive(LinesEllipsis)
  basedOn="words"
  ellipsis="..."
  innerRef={[Function]}
  maxLine="1"
  style={
    Object {
      "whiteSpace": "pre-wrap",
    }
  }
  text="Lorem ipsum dolor sit amet"
  trimRight={true}
/>
`;
